spec/federation_messages/federation_messages_spec.rb

```diff
@@ -0,0 +1,42 @@
+require "spec_helper"
+require "federation-testbed"
+
+describe "Generation and dispatch of federation messages" do
+  before :all do
+    @testbed_pod = FactoryGirl.build(:pod).host
+    FederationTestbed.config.configure(
+      "testbed_config" => {"host" => @testbed_pod, "method" => "http"},
+      "test_target"    => {
+        "method" => "http",
+        "host"   => "localhost:3000",
+        "user"   => "alice"
+      }
+    )
+    u = FederationTestbed::User.new
+    u.id = alice.diaspora_handle
+    u.public_key = alice.public_key
+    u.private_key = nil
+    FederationTestbed.test_data.push(u)
+    rack_app = FederationTestbed::App.new
+    stub_request(:get, Addressable::Template.new(@testbed_pod + "/.well-known/host-meta")).to_rack(rack_app)
+    stub_request(:get, Addressable::Template.new(@testbed_pod + "/webfinger?q=acct:test@" + @testbed_pod))
+      .to_rack(rack_app)
+    stub_request(:get, Addressable::Template.new(@testbed_pod + "/hcard/users/test@" + @testbed_pod))
+      .to_rack(rack_app)
+    stub_request(:post, Addressable::Template.new(@testbed_pod + "/receive/users/{guid}")).to_rack(rack_app)
+  end
+  before do
+    allow_any_instance_of(Postzord::Dispatcher::Private).to receive(:deliver_to_remote).and_call_original
+    allow_any_instance_of(Postzord::Dispatcher::Public).to receive(:deliver_to_remote).and_call_original
```

Comment:
Well, probably I just don't need these lines. I copied these from somewhere and thought they affect some call stubbing.